File: backend/app/api/notifications.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from backend.app.core.deps import get_current_user
from backend.app.db.session import get_db
from backend.app.models.user import User
from backend.app.repositories.notification_repo import (
    list_notifications,
    mark_all_read,
)
from backend.app.schemas.notification import NotificationOut

router = APIRouter(prefix="/notifications", tags=["notifications"])


@router.get("", response_model=list[NotificationOut])
def list_notifications_endpoint(
    unread_only: bool = False,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
) -> list[NotificationOut]:
    return list_notifications(db, user_id=current_user.id, unread_only=unread_only)


@router.post("/mark-all-read")
def mark_all_read_endpoint(
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
) -> dict:
    updated = mark_all_read(db, user_id=current_user.id)
    return {"updated": updated}

File: frontend/src/api/notifications.ts
import { apiClient } from "./client";
import type { Notification } from "../types";

// TODO: Implement API calls
export const listNotifications = async (
  unreadOnly?: boolean
): Promise<Notification[]> => {
  const params =
    typeof unreadOnly === "boolean" ? { unread_only: unreadOnly } : undefined;
  const response = await apiClient.get<Notification[]>("/notifications", {
    params,
  });
  return response.data;
};

export const markAllRead = async (): Promise<{ updated: number }> => {
  const response = await apiClient.post<{ updated: number }>(
    "/notifications/mark-all-read"
  );
  return response.data;
};
